docs/source/tutorials/datasets.rst

```diff
@@ -47,6 +47,30 @@ You could tweak :code:`max_seq_len` to achieve that directly from the config.
     # Original is 512
     max_seq_len: 256
 
+If you wish to train your model on multiple datasets and apply specific settings to each
+one, such as template, maximum sequence length, or chat format, you will need to adopt a
+different configuration format. This approach allows for a fine-grained customization of
+each dataset included in the training. Here is an example that demonstrates this flexibility:
+
+.. code-block:: yaml
+
+  # Dataset
+  dataset:
+    - _component_: torchtune.datasets.instruct_dataset
+      source: vicgalle/alpaca-gpt4
+      template: AlpacaInstructTemplate
+      split: train
+      train_on_input: True
+    - _component_: torchtune.datasets.instruct_dataset
+      source: samsum
+      template: SummarizeTemplate
+      column_map: {"output": "summary"}
+      split: train
+      train_on_input: False
+
+This configuration demonstrates how to seamlessly integrate two datasets from different
+formats for comprehensive training.
```

Comment:
Can we remove this from the live docs? I feel like there are too many subtleties here that are not immediately clear. E.g. a very natural next question based on this is "how can I sample from different datasets with different probabilities?", which is not supported. So I think it's better to leave it out until we have a full feature set, then we can showcase the functionality properly and in more detail.

Reply:
Maybe better just to change it to something like this?

```yml
  # Dataset
  dataset:
    - _component_: torchtune.datasets.alpaca_dataset
      max_seq_len: 256
    - _component_: torchtune.datasets.samsum_dataset
```

Reply:
So actually my comment is less on the config itself, but more on the way that we frame our multi-dataset support. Basically I want to make sure we are explicit about what's supported and what's not. So while the ability to configure each individual dataset separately does imply some flexibility, I would prefer not to focus on flexibility as much as you have in this particular snippet, as it can easily imply sampling strategies etc. that we do not (yet) support. Instead it'd be good to clearly state that we are concatenating the datasets when built in this way. How about something like this?

```suggestion
It is also possible to train on multiple datasets by combining them into a single :class:`~torchtune.datasets.ConcatDataset`. For example:
.. code-block:: yaml

  dataset:
    - _component_: torchtune.datasets.instruct_dataset
      source: vicgalle/alpaca-gpt4
      template: AlpacaInstructTemplate
      split: train
      train_on_input: True
    - _component_: torchtune.datasets.instruct_dataset
      source: samsum
      template: SummarizeTemplate
      column_map: {"output": "summary"}
      split: train
      train_on_input: False

The preceding snippet demonstrates how you can configure each individual dataset's parameters, then combine them into a single concatenated dataset for training.
```

(Note: for the above suggestion to work you will also need to add `ConcatDataset` to [api_ref_datasets.rst](https://github.com/pytorch/torchtune/blob/9274c8989cfa75feb8804a8e15b5c8ba9d71ccdb/docs/source/api_ref_datasets.rst#L4))

Reply:
I like it!

Thank you for pointing out `api_ref_datasets.rst`, I will address it promptly.

Reply:
`ConcatDataset` added to `api_ref_datasets.rst` as you suggested.